mm/slub.c

```diff
@@ -3151,6 +3151,11 @@ int build_detached_freelist(struct kmem_cache *s, size_t size,
 	if (!object)
 		return 0;
 
+	if (kfence_free(object)) {
+		p[size] = NULL; /* mark object processed */
```

Comment:
Hm, I think this solves the issue with `slab_free_freelist_hook()`: now it does not need to iterate the freelist for bulk allocations.

Reply:
Do you mean we can remove the KFENCE check from slab_free_freelist_hook()?

Reply:
Yes, I'll try to.

Reply:
I still don't know what the right thing here is -- I believe we definitely end up in slab_free_freelist_hook(), and the check is required. But clearly it won't work for us.

Also, we definitely need some of things inside it, e.g. 'slab_free_hook'.